pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -225,79 +225,23 @@ func (r *ReconcileWindowsMachine) addWorkerNode(ipAddress, instanceID string) er
-			$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
- 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
-			$passwordConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			$authFileConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys','#AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys'
-			$authFileConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			$pubKeyLocationConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'Match Group administrators','#Match Group administrators'
-			$pubKeyLocationConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			Restart-Service sshd
-			New-item -Path $env:USERPROFILE -Name .ssh -ItemType Directory -force
-			echo "` + string(pubKeyBytes[:]) + `"| Out-File $env:USERPROFILE\.ssh\authorized_keys -Encoding ascii
-			</powershell>
-			<persist>true</persist>`),
-		},
-	}
+// validateUserData validates userData secret. It returns error if the secret doesn`t
+// contain expected public key bytes.
+func (r *ReconcileWindowsMachine) validateUserData() error {
+	userDataSecret := &core.Secret{}
+	err := r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: "windows-user-data", Namespace: "openshift-machine-api"}, userDataSecret)
 
-	// check if the userDataSecret already exists
-	err := r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: userDataSecret.Name, Namespace: userDataSecret.Namespace}, &core.Secret{})
 	if err != nil {
-		if k8sapierrors.IsNotFound(err) {
-			log.Info("Creating a new Secret", "Secret.Namespace", userDataSecret.Namespace, "Secret.Name", userDataSecret.Name)
-			err = r.client.Create(context.TODO(), userDataSecret)
-			if err != nil {
-				return errors.Wrap(err, "error creating windows user data secret")
-			}
-		}
-		return errors.Wrap(err, "error creating windows user data secret")
+		return errors.Errorf("could not find Windows userData secret in required namespace: %v", err)
 	}
-	return nil
-}
 
-// createSigner creates a signer using the private key from the privateKeyPath
-func createSigner() (ssh.Signer, error) {
-	privateKeyBytes, err := ioutil.ReadFile(wkl.PrivateKeyPath)
+	secretData := string(userDataSecret.Data["userData"][:])
+	desiredUserDataSecret, err := secrets.GenerateUserData()
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to find private key from path: %v", wkl.PrivateKeyPath)
+		return err
 	}
-
-	signer, err := ssh.ParsePrivateKey(privateKeyBytes)
-	if err != nil {
-		return nil, errors.Wrapf(err, "unable to parse private key: %v", wkl.PrivateKeyPath)
+	if string(desiredUserDataSecret.Data["userData"][:]) != secretData {
+		return errors.Errorf("expected userData secret not present in required namespace")
```

Comment:
This seems to imply the secret is not present. It is just the contents that is invalid.